tests/api_resources/test_subscription.py

```diff
@@ -1,162 +1,137 @@
-            },
-            None
+    def test_can_delete_discount(self):
+        sub = stripe.Subscription.retrieve(TEST_RESOURCE_ID)
+        sub.delete_discount()
+        self.assert_requested(
+            'delete',
+            '/v1/subscriptions/%s/discount' % sub.id
         )
 
-    def test_modify_subscription_items(self):
-        stripe.Subscription.modify('test_sub',
-                                   items=[{"id": "si", "plan": "foo"}])
+    # Test create/modify methods with subscription items
 
-        self.requestor_mock.request.assert_called_with(
+    def test_is_creatable_with_items(self):
+        resource = stripe.Subscription.create(
+            customer='cus_123',
+            items=[{"plan": "foo", "quantity": 3}]
+        )
+        self.assert_requested(
             'post',
-            '/v1/subscriptions/test_sub',
+            '/v1/subscriptions',
             {
+                'customer': 'cus_123',
                 'items': {
                     "0": {
                         "plan": "foo",
-                        "id": "si",
+                        "quantity": 3
                     },
                 },
             },
-            None
         )
+        self.assertIsInstance(resource, stripe.Subscription)
 
-    def test_create_subscription_items(self):
-        stripe.Subscription.create(items=[{"plan": "foo", "quantity": 3}])
-        self.requestor_mock.request.assert_called_with(
+    def test_is_modifiable_with_items(self):
+        resource = stripe.Subscription.modify(
+            TEST_RESOURCE_ID,
+            items=[{"id": "si", "plan": "foo"}]
+        )
+        self.assert_requested(
             'post',
-            '/v1/subscriptions',
+            '/v1/subscriptions/%s' % TEST_RESOURCE_ID,
             {
                 'items': {
                     "0": {
                         "plan": "foo",
-                        "quantity": 3,
+                        "id": "si"
                     },
                 },
             },
-            None
-        )
-
-    def test_delete_subscription(self):
-        subscription = stripe.Subscription.construct_from({
-            'id': 'test_sub',
-            'customer': 'test_cus',
-        }, 'api_key')
-
-        subscription.delete()
-
-        self.requestor_mock.request.assert_called_with(
-            'delete',
-            '/v1/subscriptions/test_sub',
-            {},
-            None
-        )
-
-    def test_delete_subscription_discount(self):
-        subscription = stripe.Subscription.construct_from({
-            'id': 'test_sub',
-            'customer': 'test_cus',
-            'coupon': 'test_discount'
-        }, 'api_key')
-
-        subscription.delete_discount()
-
-        self.requestor_mock.request.assert_called_with(
-            'delete',
-            '/v1/subscriptions/test_sub/discount'
         )
+        self.assertIsInstance(resource, stripe.Subscription)
+
+    # TODO: Fix this test
```

Comment:
@ob-stripe Flagging this one because I don't understand it. The error is 
```
======================================================================
FAIL: test_is_saveable_with_items (tests.api_resources.test_subscription.SubscriptionTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/remi/Workspace/bindings/stripe-python/tests/api_resources/test_subscription.py", line 129, in test_is_saveable_with_items
    "id": "si"
  File "/Users/remi/Workspace/bindings/stripe-python/tests/helper.py", line 92, in assert_requested
    return self.request_mock.assert_requested(*args, **kwargs)
  File "/Users/remi/Workspace/bindings/stripe-python/tests/request_mock.py", line 55, in assert_requested
    raise exception
AssertionError: Expected call: request(<ANY>, 'post', '/v1/subscriptions/sub_123', {'items': {'0': {'plan': 'foo', 'id': 'si'}}}, <ANY>)
Actual call: request(<stripe.api_requestor.APIRequestor object at 0x10dca58d0>, 'post', '/v1/subscriptions/sub_BQp6MLuprKbRgM', {u'items': {'0': {'id': 'si', 'plan': 'foo'}}, u'metadata': {}}, None)
```

There seems to be:
* an encoding issue?
* metadata being there when I'm not sure it should?

Reply:
> an encoding issue?

Do you mean the `items` key string being converted to Unicode? That shouldn't be an issue as it doesn't affect string comparison:
```python
>>> 'foo' == u'foo'
True
```

> metadata being there when I'm not sure it should?

It's expected that `metadata` is present here. It won't be sent in the actual request (the value is an empty dict, so it will just be ignored when the parameters are encoded).

Ideally, `metatadata` should _not_ be in the parameters (even though it has no impact on functionality, it's annoying for our tests). #372 seems to take care of that though! I still need to take a closer look but it looks promising.